### PR TITLE
NMS-14665:Dynamic Yaml generation: Duplicate build entry when we enable flaky test and doc jobs

### PR DESCRIPTION
**Pull request by mershad-manesh**
We ran into an intermittent issue where processing the job dependencies would create an invalid YAML file.

In this PR, we are changing the logic used to generate the YAML file. 

Before this PR, we calculated the dependencies as we went through the job.

In the new approach, we will go through the list of dependencies for each job and figure out their requirements. Then we process them one by one. 

### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14665

### CODE REVIEW

**Comment by RangerRick** [comment]
Only comment I have so far is that it looks like you reformatted the code, and it's really hard to tell the "real" fixes from the noise from reformatting; could you rebase this down to something that makes it easier to review?

**Review by RangerRick — DISMISSED** [review]
a `git diff -b -w` made this easier to follow, seems OK to me

File: .circleci/pyscripts/generate_main.py [hunk 1]
```diff
@@ -26,13 +27,13 @@ def combine_workflow_path(job_entry_spaces, workflow_path):
     _entries = []
     combined_entries = []
     for index, element in enumerate(workflow_path):
-        re_match = re.match("^"+(" "*job_entry_spaces)+"- ", element)
+        re_match = re.match("^" + (" " * job_entry_spaces) + "- ", element)
```
Comment (bborovan) [inline .circleci/pyscripts/generate_main.py:30]: You might want to consider using Python f-strings to make this cleaner to read:
https://docs.python.org/3/whatsnew/3.6.html#whatsnew36-pep498 

Not absolutely required, but just a note.

**Review by RangerRick — APPROVED** [review]
yes!

**Pull request merged**